### PR TITLE
fixed the duplicate code by adding functions

### PR DESCRIPTION
**Pull request by Sanket-0510**
Fixed duplicate code by replacing it using functions, kindly guide me if I am on right direction or not.

### CODE REVIEW

**Comment by xermicus** [comment]
Can you please merge the upstream main into your branch?

File: src/emit/substrate/target.rs [hunk 1]
```diff
@@ -361,14 +361,20 @@ impl<'a> TargetRuntime<'a> for SubstrateTarget {
...
+               binary.builder.build_gep(
+               &binary.context.i8_type().array_type(SCRATCH_SIZE),
+               &binary.scratch.unwrap().as_pointer_value(),
+               &[i32_zero!(),index],
+               "data_offset",
             )
-        };
+
+            }
+        }
```
Comment (xermicus) [inline src/emit/substrate/target.rs:374]: Formatting seemse to be off

**Comment by Sanket-0510** [comment]
I have updated the branch, kindly tell me if further more changes are required

File: src/emit/substrate/target.rs [hunk 2]
```diff
@@ -361,14 +361,20 @@ impl<'a> TargetRuntime<'a> for SubstrateTarget {
 
         binary.builder.position_at_end(retrieve_block);
 
-        let offset = unsafe {
-            binary.builder.build_gep(
-                binary.context.i8_type().array_type(SCRATCH_SIZE),
-                binary.scratch.unwrap().as_pointer_value(),
-                &[i32_zero!(), index],
-                "data_offset",
+        fn function_to_get_pointer(binary: &Binary, index: &IntValue)-> PointerValue{
```
Comment (LucasSte) [inline src/emit/substrate/target.rs:364]: This name is not good. The operation here indexes the scratch array, not only returns a pointer to something. The name should describe what this function does.

**Review by LucasSte — CHANGES_REQUESTED** [review]

**Comment by LucasSte** [comment]
@Sanket-0510 Please, run `cargo test` locally to test your changes. In addition, run `cargo fmt --all` to format your code and `cargo clippy --workspace --tests --bins -- -D warnings -D clippy::inconsistent-struct-constructor` to verify proper Rust conventions.

File: src/emit/substrate/target.rs (hunk 2 above)
Comment (Sanket-0510) [reply src/emit/substrate/target.rs:364]: got it I will surely try to work upon this, thank you so much for your input.

**Comment by Sanket-0510** [comment]
surely I will do before opening any pr from now onwards

**Comment by xermicus** [comment]
I gave your PR for a CI run - please see the comment by Lucas on how you can test some of the steps locally. Also note that the build fails.

**Comment by xermicus** [comment]
This PR has been stale for over a month; if you still want to finish it please just ping me here

**Pull request closed without merging**